src/checkers/inference/InferenceQualifierHierarchy.java

```diff
@@ -307,7 +273,7 @@ public AnnotationMirror getTopAnnotation(final AnnotationMirror am) {
             return varAnnot;
         } //else
 
-        return unqualified;
+        return inferenceMain.getRealTypeFactory().getQualifierHierarchy().getTopAnnotations().iterator().next();
```

Comment:
Is this branch needed? From other changes it looks like only VarAnnot subtypes are expected.